bing_bert/nvidia/modeling.py

```diff
@@ -886,14 +886,17 @@ class BertForPreTraining(BertPreTrainedModel):
     def __init__(self, config, args):
         super(BertForPreTraining, self).__init__(config)
         self.summary_writer = None
-        if dist.get_rank() == 0:
-            self.summary_writer = args.summary_writer
-        self.samples_per_step = dist.get_world_size() * args.train_batch_size
-        self.sample_count = self.samples_per_step
         self.bert = BertModel(config)
         self.cls = BertPreTrainingHeads(config, self.bert.embeddings.word_embeddings.weight)
         self.apply(self.init_bert_weights)
 
+    def set_summary_writer(self, summary_writer):
+        self.summary_writer = summary_writer
+
+    def set_samples_per_step(self, train_batch_size):
+        self.samples_per_step = dist.get_world_size() * train_batch_size
+        self.sample_count = self.samples_per_step
```

Comment:
why are there two members that have the same value?

Reply:
samples per step never changes but sample count will grow with each step, see line 906

Reply:
I see. What if log_summary_writer is not called every step? Does it make sense to keep track of samples_per_step and sample_count inside deepspeed or maybe even in the client code if we don't want it in deepspeed and increment it every step?

Reply:
Originally this was happening with every every forward pass. However, after further inspection it appears it's not even used in this code currently: https://github.com/microsoft/DeepSpeedExamples/blob/master/bing_bert/nvidia/modeling.py#L925

Reply:
I was just about to ask where log_summary_writer is actually used

Reply:
Will push a commit to remove all of this now, there's _tons_ of dead code spread throughout this model... :(

Reply:
sounds good

Reply:
Code removed, take a look at the new commit